### PR TITLE
Update JSON schema in CNB Registry

### PR DESCRIPTION
**Pull request by jkutner**
[Readable](https://github.com/buildpacks/rfcs/blob/update-json-cnb-registry/text/0000-update-json-cnb-registry.md)

### CODE REVIEW

File: text/0000-update-json-cnb-registry.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+# How it Works
+[how-it-works]: #how-it-works
+
+An entry will have the following structure:
+
+```
+{
+  "ns" : "<string>",
+  "name": "<string>",
+  "version" : "<string",
```
Comment (jromero) [inline text/0000-update-json-cnb-registry.md]:
```suggestion
  "version" : "<string>",
```

File: text/0000-update-json-cnb-registry.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+An example of what this may look like for a single buildpack file:
+
+```
+{"ns":"heroku","name":"ruby","version":"0.1.0","yanked":false,"addr":"docker.io/hone/ruby-buildpack@sha256:a9d9038c0cdbb9f3b024aaf4b8ae4f894ea8288ad0c3bf057d1157c74601b906"}
+{"ns":"heroku","name":"ruby","version":"0.2.0","yanked":false,"addr":"docker.io/hone/ruby-buildpack@sha256:2560f05307e8de9d830f144d09556e19dd1eb7d928aee900ed02208ae9727e7a"}
+{"ns":"heroku","name":"ruby","version":"0.2.1","yanked":false,"addr":"docker.io/hone/ruby-buildpack@sha256:74eb48882e835d8767f62940d453eb96ed2737de3a16573881dcea7dea769df7"}
+{"ns":"heroku","name":"ruby","version":"0.3.0","yanked":false,"addr":"docker.io/hone/ruby-buildpack@sha256:8c27fe111c11b722081701dfed3bd55e039b9ce92865473cf4cdfa918071c566"}
+```
+
+*Note:* id is the combination of two fields, `ns` and `name`. The `/` will be replaced by a `_` in the filename. The directories will use the first four characaters of the buildpacks `name`, split into two two character dirs. For example:
```
Comment (jromero) [inline text/0000-update-json-cnb-registry.md]: Based on some of the other comments in https://github.com/buildpacks/rfcs/pull/35.

Moving to using the buildpack name for the directory structure seems like a good idea for improving the sharding. It does seem like discoverability/troubleshooting would be impacted. I wonder if the original alternative proposal named "Namespace as top-level dirs" would suite both requirements better.

```
heroku
└── ja
    ├── va
    │   ├── heroku_java
    │   └── heroku_javascript
    └── co
        └── heroku_jacoco
```

Comment (jromero) [reply text/0000-update-json-cnb-registry.md]: Adding a more detailed examples:

##### without ns

```
ja
├── va
│   ├── example_java
│   ├── example_javascript
│   ├── heroku_java
│   └── heroku_javascript
└── co
    └── heroku_jacoco
```

##### with ns

```
heroku
└── ja
    ├── va
    │   ├── heroku_java
    │   └── heroku_javascript
    └── co
        └── heroku_jacoco
example
└── ja
    └── va
        ├── example_java
        └── example_javascript

```

**Comment by nebhale** [comment]
My vote is for hashing based on buildpack name not namespace name.

File: text/0000-update-json-cnb-registry.md (hunk 2 above)
Comment (jromero) [reply text/0000-update-json-cnb-registry.md]: Based on the context/history provided in the working group meeting I would consider this conversation resolved.

**Comment by hone** [comment]
@jkutner you need to fix the DCO of the suggested fix by Javier.

**Pull request merged**